ast/term.go

```diff
@@ -2063,6 +2028,42 @@ func (obj *object) insert(k, v *Term) {
 	}
 }
 
+func (obj *object) remove(k *Term) {
+	hash := k.Hash()
+	head := obj.elems[hash]
+	equal := keyEqual(k)
+
+	var prev *objectElem = nil
+	for curr := head; curr != nil; curr = curr.next {
+		if equal(curr.key.Value) {
+			if curr.value.IsGround() {
+				obj.ground--
+			}
+
+			if prev == nil {
+				obj.elems[hash] = curr.next
+			} else {
+				prev.next = curr.next
+			}
+		} else {
+			prev = curr
+		}
+	}
```

Comment:
Just for my understanding, under which circumstances would more than one key of an object be equal to `k`?

Reply:
The code was a bit easier to write that way I think, but this no longer applies in the new diff anyway.